lib/r_sig.rb

```
@@ -0,0 +1,21 @@
+# typed: true
+
+module RSig
+  T::Sig::WithoutRuntime.sig { params(ctx: Module, blk: T.proc.bind(T::Private::Methods::DeclBuilder).void).void }
+  def self.sig(ctx, &blk)
```

RSig.sig is not bad, but it introduces another top-level module that could potentially conflict with user code or other gems, even if unlikely.

Instead, would it be possible to extend `RSpec::Core::ExampleGroup` with an `rsig` method?

---

I can look into that, it may make the ast parsing more complicated or error-prone though, as then you have the possibility of a name collision with an `rsig` method defined in any namespace in user code, and if that other `rsig` method is used in a spec file, there may be a false positive. Whereas in this case the name collision would be limited to the single case of `RSig#sig` specifically. Though it may be trivial to prevent issues even in those edge cases, by having a rescue block around the sorbet internal sig compilation step and silently skipping like it already does when other unexpected anomalies are detected.

As a side note, if we are open to patching rspec itself, there may be a way to avoid using ast parsing entirely. I would love to have a simpler way to accomplish what we're doing here, but it may prove to be complex in other ways if going that route. 

What would be even better is if there was a way for a standard sorbet sig to be definable without being subject to the "no two sigs in a row" restriction, which is why we can't use them.